shm/src/ship.cpp

```diff
@@ -5,16 +5,26 @@
 
 Ship::Ship() : id_(-1) {}
 
-Ship::Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id, Delegate* delegate)
-    : delegate_(delegate), 
+Ship::Ship(int capacity,
+           int maxCrew,
+           int speed,
+           const std::string& name,
+           size_t id,
+           Delegate* delegate,
+           Time* time)
+    : delegate_(delegate),
       capacity_(capacity),
       maxCrew_(maxCrew),
       crew_(0),
       speed_(speed),
       name_(name),
-      id_(id) {}
+      id_(id),
+      time_(time) {}
 
-Ship::Ship(int maxCrew, int speed, size_t id, Delegate* delegate) : Ship(0, maxCrew, speed, "", id, delegate) {}
+Ship::Ship(int maxCrew, int speed, size_t id, Delegate* delegate, Time* time)
+    : Ship(0, maxCrew, speed, "", id, delegate, time) {
+    time_->addObserver(this);
+}
```

Comment:
I don't know if it is correctly because now you can't make object for example with nonzero capacity and with added observer.

Reply:
You are right, it not correctly made. 
I have moved time_->addObserver(this) to the first constructor.